### ISSUE TITLE
unit/integration flake: Connection timed out error in test-cmd.sh

### ISSUE THREAD

**Issue by david-mcmahon** [kubernetes/kubernetes#23747]
```
unable to read URL "https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/testdata/multi-resource-yaml.yaml", server reported 503 503 Connection timed out
!!! Error in ./hack/test-cmd.sh:1554
  'kubectl create -f https://raw.githubusercontent.com/kubernetes/kubernetes/master/hack/testdata/multi-resource-yaml.yaml "${kube_flags[@]}"' exited with status 1

Ref #23452 
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/23452/kubernetes-pull-test-unit-integration/19892/
```

**Comment by ixdy** [comment]
We've had a lot of flakiness pulling down files from githubusercontent.com; usually we've addressed this by adding `--retry` to `curl` calls. Does `kubectl` have a retry option?

cc @kubernetes/kubectl

**Comment by lavalamp** [comment]
@pwittrock @janetkuo are the closest thing we have to kubectl owners.

**Comment by smarterclayton** [comment]
If possible I'd like to make retry in the CLI very, very constrained and be
deliberate about it.  We had lots of little flakes in reapers and labels
and annotations that we fixed by doing targeted retry automatically - we
could potentially do the same for URL downloads in resource builder without
making it explicit.

**Comment by pwittrock** [comment]
retrying to download the url seems reasonable.

**Comment by pwittrock** [comment]
Should be resolved in #25267
